Intersect.Client.Core/Interface/Game/Character/CharacterWindow.cs

```diff
@@ -411,29 +383,19 @@ public void UpdateExtraBuffs()
         mPlayer = ClassDescriptor.Get(Globals.Me?.Class ?? Guid.Empty);
 
         //Getting HP and Mana Regen
-        if (mPlayer != null)
-        {
-            HpRegenAmount = mPlayer.VitalRegen[0];
-            mHpRegen.SetText(Strings.Character.HealthRegen.ToString(HpRegenAmount));
-            ManaRegenAmount = mPlayer.VitalRegen[1];
-            mManaRegen.SetText(Strings.Character.ManaRegen.ToString(ManaRegenAmount));
-        }
+        HpRegenAmount = mPlayer?.VitalRegen[0] ?? 0;
+        ManaRegenAmount = mPlayer?.VitalRegen[1] ?? 0;
 
-        CooldownAmount = 0;
+        //Reset all values
         LifeStealAmount = 0;
```

Comment:
If the update only happens when hovering, then would all these properties still be necessary? If not, would there be more code to remove, simplifying the window further?

Reply:
yes, we actually need these so the values aren't endleslly added